ui/app/AppLayouts/Communities/panels/SharedAddressesAccountSelector.qml

```diff
@@ -0,0 +1,185 @@
+
+StatusListView {
+    id: root
+
+    property bool hasPermissions
+    property var uniquePermissionTokenKeys
+
+    // read/write properties
+    property string selectedAirdropAddress
+    property var selectedSharedAddresses: []
+
+    leftMargin: d.absLeftMargin
+    topMargin: Style.current.padding
+    rightMargin: Style.current.padding
+    bottomMargin: Style.current.padding
+
+    Instantiator {
+        id: selectionChecker
+
+        model: root.model
+
+        property var keys: []
+
+        delegate: QtObject {
+            readonly property string key: model.address
+        }
+
+        onObjectAdded: {
+            const keysCopy = [...keys]
+            keysCopy.splice(index, 0, object.key)
+            keys = keysCopy
+        }
+
+        onObjectRemoved: {
+            const keysCopy = [...keys]
+            keysCopy.splice(index, 1)
+            keys = keysCopy
+        }
+    }
+
+    Component.onCompleted: {
+        if (!root.selectedSharedAddresses.length)
+            root.selectedSharedAddresses = selectionChecker.keys // pre-select all by default
+        if (!root.selectedAirdropAddress)
+            root.selectedAirdropAddress = root.selectedSharedAddresses[0] // pick the first one
+    }
+
+    QtObject {
+        id: d
+
+        // UI
+        readonly property int absLeftMargin: 12
+
+        readonly property var airdropGroup: ButtonGroup {
+            exclusive: true
+        }
+
+        readonly property var addressesGroup: ButtonGroup {
+            exclusive: false
+        }
+
+        function firstVisibleAirdropButton() {
+            for (let i = 0; i < airdropGroup.buttons.length; i++) {
+                const button = airdropGroup.buttons[i]
+                if (button.visible)
+                    return button
+            }
+        }
+    }
+
+    spacing: Style.current.halfPadding
+    delegate: StatusListItem {
+        width: ListView.view.width - ListView.view.leftMargin - ListView.view.rightMargin
+        statusListItemTitle.font.weight: Font.Medium
+        title: model.name
+        tertiaryTitle: !walletAccountAssetsModel.count && root.hasPermissions ? qsTr("No relevant tokens") : ""
+
+        readonly property var walletAccountAssetsModel: SortFilterProxyModel {
+            id: walletAccountAssetsModel
+            sourceModel: model.assets
+
+            function filterPredicate(modelData) {
+                return root.uniquePermissionTokenKeys.includes(modelData.symbol.toUpperCase())
+            }
+
+            filters: ExpressionFilter {
+                expression: walletAccountAssetsModel.filterPredicate(model)
+            }
+            sorters: ExpressionSorter {
+                expression: {
+                    return modelLeft.enabledNetworkBalance.amount > modelRight.enabledNetworkBalance.amount // descending, biggest first
+                }
+            }
+        }
+
+        tagsModel: SortFilterProxyModel {
+            filters: IndexFilter {
```

Comment:
Please revert https://github.com/status-im/SortFilterProxyModel/pull/1 as it's (according to our discussion) unreliable and unused.

Reply:
I didn't use it here; I'll do a separate PR to revert the last commit there

Reply:
https://github.com/status-im/SortFilterProxyModel/pull/2